lib/linguist/heuristics.rb

```diff
@@ -274,7 +274,7 @@ def call(data)
     end
 
     disambiguate ".pl" do |data|
-      if /^(use v6|(my )?class|module)/.match(data)
+      if /^(use v6|(my)? class|module)/.match(data) #class might be a variable name in Perl 5
```

Comment:
2015-11-01 11:25 GMT+01:00 Paul Chaignon <notifications@github.com>:

> In lib/linguist/heuristics.rb
> <https://github.com/github/linguist/pull/2709#discussion_r43581107>:
>
> > @@ -274,7 +274,7 @@ def call(data)
> >      end
> >
> >      disambiguate ".pl" do |data|
> > -      if /^(use v6|(my )?class|module)/.match(data)
> > +      if /^(use v6|(my)? class|module)/.match(data) #class might be a variable name in Perl 5
>
> Could you add a test for this? I don't really understand the reasoning
> behind this :/
>

When you have a variable called $class it gives a false positive. Yep, I
guess I can.